### PR TITLE
testdrive: add a new `psql-execute` action

### PR DESCRIPTION
**Pull request by benesch**
This command will be useful for writing integration tests for the
various backslash commands in psql. This commit adds only documentation
and one sample use of the action.

### Motivation

  * This PR adds a feature that has not yet been specified.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any
  [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

**Review by philip-stoev — APPROVED** [review]
Thanks a lot! This may end up being more useful than `postgres-execute` as it allows you to assert against the expected output.

That said, may I interest you in my upcoming book by O'Reilly:
leave-testdrive-to-the-grunts

File: src/testdrive/src/util/text.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+use atty::Stream;
+use similar::{ChangeTag, TextDiff};
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+/// Trims trailing whitespace from each line of `s`.
+pub fn trim_trailing_space(s: &str) -> String {
+    let mut lines: Vec<_> = s.lines().map(|line| line.trim_end()).collect();
+    while lines.last().map_or(false, |l| l.is_empty()) {
+        lines.pop();
+    }
```
Comment (mkysel) [inline src/testdrive/src/util/text.rs:19]: does this remove empty lines?

**Review by mkysel — APPROVED** [review]

**Comment by mkysel** [comment]
@philip-stoev wow! That image (really, a meme!) is killer. Did you make it by hand or did someone come up with a `O'rly` book front page generator :)?

File: src/testdrive/src/util/text.rs (hunk 1 above)
Comment (benesch) [reply src/testdrive/src/util/text.rs:19]: Yeah, it removes trailing lines that consist of only whitespace.

**Comment by benesch** [comment]
@philip-stoev I see your O'Reilly book and raise you my upcoming O'Reilly book:

generate

**Pull request merged**